### ISSUE TITLE
Simplify installation process (move metadata to separate file)

### ISSUE THREAD

**Issue by kuszaj** [PlasmaPy/PlasmaPy#75]
Currently the installation process requires 2 steps, one of which is installing required packages by running `pip install -r <requirements file>` command. I think this should be handle automatically by `setup.py`, without forcing users to do it themselves. Moreover, this method might be a problem in the future, when PlasmaPy will become mature enough for its first PyPI release.

Unfortunately, with right now `setup.py` imports the whole package and from it extracts necessary package metadata: its name, current version and description. Because of it, running `python setup.py install` before installing requirements ends with a `ModuleNotFoundError`:

```shell
$ python setup.py install
Traceback (most recent call last):
  File "setup.py", line 2, in <module>
    import plasmapy
  File "/path/to/PlasmaPy/plasmapy/__init__.py", line 7, in <module>
    from .classes import Plasma
  File "/path/to/PlasmaPy/plasmapy/classes/__init__.py", line 1, in <module>
    from .plasma import Plasma
  File "/path/to/PlasmaPy/plasmapy/classes/plasma.py", line 8, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
```
Since `setup.py` needs only `PlasmaPy` metadata (i.e. name, version and description), I propose we move it to a new file `plasmapy/_metadata.py`, which could be used by both `plasmapy/__init__.py` and `setup.py`. This would bring a couple of improvements:

1. We would have a single clearly defined place, where those values would be defined. This is even more important when taking the fact that currently the package itself doesn't contain its own version, which should be accessible by e.g. invoking `plasmapy.__version__` (therefore compliant with [PEP 396](https://www.python.org/dev/peps/pep-0396/)).
2. Installation process would be limited to only one command (either `pip install .` from the top directory or `python setup.py install`). No more `ModuleNotFoundError`s.
3. Stable releases would be made easier (I'm not sure if currently releasing PlasmaPy on PyPI would be even possible).
4. Limit the number of future changes done to `setup.py`.

Additionally, I think we should also change the way of how requirements are defined in `setup.py`. Since PlasmaPy already has `requirements/base.txt`, why not just make `setup.py` read it and extract package requirements from it automatically? That way any future changes to requirements would be limited to just `requirements/base.txt` file.

If no one has any objections, I would like to prepare necessary changes, it shouldn't be much of a problem.

**Comment by StanczakDominik** [comment]
That sounds awesome! I don't think any of us have much experience with packagin, so if you could try to do that, it'd be amazing!

**Comment by kuszaj** [comment]
Unfortunately neither have I, however after my changes in `setup.py` I was able to make both source distribution (`*.tar.gz`) and a wheel (`*.whl`). Both successfully installed on a fresh virtual environment, so I think all should be fine.

**Comment by SolarDrew** [comment]
So the installation process is a bit under-developed at the moment, mostly because we haven't had an actual release yet. I've got things set up so that when we hit 0.1 I can put it up on PyPI and after that you'll be able to do `pip install plasmapy` and it should just work. It would be great to have a properly working `setup.py` though, because I guess some people will want to install manually, so thanks :D

**Comment by StanczakDominik** [comment]
Closed by #76 :)
